### PR TITLE
Create EmployeeSalaryDetails Entity & Crud

### PR DESCRIPTION
**Pull request by Lucien-vanWyk**
I added code to create the entity and crud for the new EmployeeSalaryDetails. As discussed, I will complete my Unit Tests after this PR. Bellow are screenshots of the functioning endpoints.

### Controller:

image

### Adding User Salary:

image
image

### Get All Salaries:

image
image

### Get User Salary:

image
image

### Update User Salary:

image
image

### Delete User Salary:

image
image

### CODE REVIEW

File: RR.UnitOfWork/DatabaseContext.cs [hunk 1]
```diff
@@ -37,6 +38,7 @@ public interface IDatabaseContext
     DbSet<ErrorLogging> errorLogging { get; set; }
     DbSet<Candidate> candidate { get; set; }
     DbSet<WorkExperience> workExperience { get; set; }
+
```
Comment (LlewellynAnthony) [inline RR.UnitOfWork/DatabaseContext.cs:41]:
```suggestion
```

File: HRIS.Services.Tests/Services/EmployeeSalaryDetailsServiceUnitTest.cs [hunk 2]
```diff
@@ -0,0 +1,169 @@
+﻿//TODO: Complete Unit Tests
+
```
Comment (tbddorothy) [inline HRIS.Services.Tests/Services/EmployeeSalaryDetailsServiceUnitTest.cs:2]: Absolutely not, remove this file and put the tests in a different PR

File: RR.App.Tests/Controllers/HRIS/EmployeeSalaryDetailsControllerUnitTest.cs [hunk 3]
```diff
@@ -0,0 +1,288 @@
+﻿//TODO: Complete Unit Tests
```
Comment (tbddorothy) [inline RR.App.Tests/Controllers/HRIS/EmployeeSalaryDetailsControllerUnitTest.cs:1]: Bruh.

File: HRIS.Services.Tests/Services/EmployeeSalaryDetailsServiceUnitTest.cs (hunk 2 above)
Comment (Lucien-vanWyk) [reply HRIS.Services.Tests/Services/EmployeeSalaryDetailsServiceUnitTest.cs]: Apologies, I removed the unit tests and added them to a separate branch

File: RR.App.Tests/Controllers/HRIS/EmployeeSalaryDetailsControllerUnitTest.cs (hunk 3 above)
Comment (Lucien-vanWyk) [reply RR.App.Tests/Controllers/HRIS/EmployeeSalaryDetailsControllerUnitTest.cs]: Apologies, I removed the unit tests and added them to a separate branch

**Review by tbddorothy — APPROVED** [review]

**Review by MatthewSchoeman — APPROVED** [review]
